### ISSUE TITLE
How to train yolo11-cls for single class classification?

### ISSUE THREAD

**Issue by Shehjad-Ishan** [ultralytics/ultralytics#17446]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have about 200,000 images on average 50x50 in size. I want to train a yolo11-cls model. First, I tried to train yolo11m-cls model with the default values for hyperparameters. But even after 30 epochs loss was 1. I tried to infer with the model always predicting the same label. Then I tried with lr0= 0.0001. Seemed like the model wasn't learning. Then I tried to train yolo11n-cls. Now, throughout 15 epochs the loss remained 0. Is it possible to train a single class classifier with such small images? What default values for parameters should I change?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Shehjad-Ishan, thank you for your question on training YOLO11 for single class classification! 🚀 This is an automated response to assist you while an Ultralytics engineer reviews your issue.

For new users, we suggest visiting our [Docs](https://docs.ultralytics.com) where you can find a range of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that might address some of your questions.

Given this is a query about custom training, please provide a few more details, such as dataset images examples and training logs, to help us understand the problem better. Also, ensure that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 

When dealing with a potential 🐛 bug, providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) will greatly assist in the debugging process.

Upgrade your `ultralytics` package with all required dependencies to make sure you are using the latest features:

```bash
pip install -U ultralytics
```

For real-time support and to become part of the Ultralytics community, join us on [Discord](https://ultralytics.com/discord) 🎧. For more detailed discussions, visit our [Discourse](https://community.ultralytics.com), or engage in threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Environments

Consider using one of the verified environments for running YOLO, which comes with all dependencies preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests ensure the correct operation across all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Let's see what insights the community and the Ultralytics team have to enhance your YOLO11 training experience! 🚀

**Comment by Y-T-G** [comment]
Single class classification doesn't make any sense. Classification requires at least two classes, even if it's the positive class vs. everything else.

**Comment by Y-T-G** [comment]
Classification requires at least two classes. For example, "cat" vs. "not cat" would still be two classes.

**Comment by Shehjad-Ishan** [comment]
> Classification requires at least two classes. For example, "cat" vs. "not cat" would still be two classes. You can't just have a "cat" class. The model has to also learn what's "not cat".

Will the negative class require to be as large as the positive class?

**Comment by Y-T-G** [comment]
Yes

**Comment by Shehjad-Ishan** [comment]
> Yes

This time I trained yolo11n-cls with 2 classes. When the loss decreased to 0.002 I tested the model. It's showing 2 labels where levels should be 1.

**Comment by Y-T-G** [comment]
> It's showing 2 labels where labels should be 1.

What do you mean?

**Comment by Shehjad-Ishan** [comment]
> > It's showing 2 labels where labels should be 1.
> 
> What do you mean?

I inferenced on an image of class 0. Nothing else in the image. Result showing both labels. 0 and 1.

**Comment by Y-T-G** [comment]
Is this ultralytics inference? What's the code?

**Comment by Shehjad-Ishan** [comment]
> Is this ultralytics inference? What's the code?

`model = YOLO("D:/Drone_data_inference/runs/classify/train21/weights/best.pt")`
`results = model.train(data="D:/Drone_data_inference/datasets/", epochs=150, imgsz=640, amp= True, batch=70, val=False, resume= True, pretrained= True, lr0=0.0001, optimizer= 'Adam')`

`      Epoch    GPU_mem       loss  Instances       Size
     24/150      7.51G   0.002771         70        640:   2%|▏         | 123/5622 [01:01<45:56,  1.99it/s]  `

**Comment by Y-T-G** [comment]
By default, the plot will show top 5 classes. If you want to find the best class, you can use `result.probs.top1`.

**Comment by Shehjad-Ishan** [comment]
> result.probs.top1

I see. Thank you.
